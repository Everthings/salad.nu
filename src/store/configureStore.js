import { configureStore, getDefaultMiddleware } from "@reduxjs/toolkit";
import rootReducer from "./reducer";
import api from "./middleware/api";

export default function () {
  return configureStore({
    reducer: rootReducer,
    middleware: [...getDefaultMiddleware(), api],
  });
}
