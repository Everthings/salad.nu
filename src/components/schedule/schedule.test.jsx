import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import configureStore from "./../../store/configureStore";
import { Provider } from "react-redux";
import { ToastProvider } from "react-toast-notifications";
import Theme from "./../themes/theme";
import Schedule from "./schedule";

ReactDOM.createPortal = (node) => node;

describe("Schedule", () => {
  it("renders schedule correctly", () => {
    const store = configureStore();
    const tree = renderer
      .create(
        <ToastProvider>
          <Provider store={store}>
            <Theme>
              <Schedule />
            </Theme>
          </Provider>
        </ToastProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders schedule correctly with hovered section", () => {
    const initialState = {
      interactions: {
        hoveredScheduledSection: { id: 0 },
        hoveredSection: { id: 1 },
      },
      sections: {
        list: [{ unique_id: 1, start_time: "1:00", end_time: "2:00" }],
      },
    };
    const store = configureStore(initialState);
    const tree = renderer
      .create(
        <ToastProvider>
          <Provider store={store}>
            <Theme>
              <Schedule />
            </Theme>
          </Provider>
        </ToastProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
