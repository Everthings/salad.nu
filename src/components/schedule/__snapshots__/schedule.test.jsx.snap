// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule renders schedule correctly with hovered section 1`] = `
Array [
  <div
    className="sc-fznKkj jOsktr"
  >
    <div
      className="sc-fznZeY fKZSHG"
    >
      <div
        className="sc-AxjAm dzsSRM"
      >
        <div
          className="sc-AxirZ bzvTeL"
        >
          <div
            className="sc-AxiKw WupAO"
          >
            1 am
          </div>
        </div>
        <div
          className="sc-AxirZ bzvTeL"
        >
          <div
            className="sc-AxiKw WupAO"
          >
            2 am
          </div>
        </div>
        <div
          className="sc-AxirZ bzvTeL"
        >
          <div
            className="sc-AxiKw WupAO"
          >
            3 am
          </div>
        </div>
        <div
          className="sc-AxirZ bzvTeL"
        >
          <div
            className="sc-AxiKw WupAO"
          >
            4 am
          </div>
        </div>
        <div
          className="sc-AxirZ bzvTeL"
        >
          <div
            className="sc-AxiKw WupAO"
          >
            5 am
          </div>
        </div>
        <div
          className="sc-AxirZ bzvTeL"
        >
          <div
            className="sc-AxiKw WupAO"
          >
            6 am
          </div>
        </div>
        <div
          className="sc-AxirZ bzvTeL"
        >
          <div
            className="sc-AxiKw WupAO"
          >
            7 am
          </div>
        </div>
        <div
          className="sc-AxirZ bzvTeL"
        >
          <div
            className="sc-AxiKw WupAO"
          >
            8 am
          </div>
        </div>
        <div
          className="sc-AxirZ bzvTeL"
        >
          <div
            className="sc-AxiKw WupAO"
          >
            9 am
          </div>
        </div>
        <div
          className="sc-AxirZ bzvTeL"
        >
          <div
            className="sc-AxiKw WupAO"
          >
            10 am
          </div>
        </div>
        <div
          className="sc-AxirZ bzvTeL"
        >
          <div
            className="sc-AxiKw WupAO"
          >
            11 am
          </div>
        </div>
        <div
          className="sc-AxirZ bzvTeL"
        >
          <div
            className="sc-AxiKw WupAO"
          >
            12 pm
          </div>
        </div>
        <div
          className="sc-AxirZ bzvTeL"
        >
          <div
            className="sc-AxiKw WupAO"
          >
            1 pm
          </div>
        </div>
        <div
          className="sc-AxirZ bzvTeL"
        >
          <div
            className="sc-AxiKw WupAO"
          >
            2 pm
          </div>
        </div>
        <div
          className="sc-AxirZ bzvTeL"
        >
          <div
            className="sc-AxiKw WupAO"
          >
            3 pm
          </div>
        </div>
        <div
          className="sc-AxirZ bzvTeL"
        >
          <div
            className="sc-AxiKw WupAO"
          >
            4 pm
          </div>
        </div>
        <div
          className="sc-AxirZ bzvTeL"
        >
          <div
            className="sc-AxiKw WupAO"
          >
            5 pm
          </div>
        </div>
        <div
          className="sc-AxirZ bzvTeL"
        >
          <div
            className="sc-AxiKw WupAO"
          >
            6 pm
          </div>
        </div>
      </div>
      <div
        className="sc-fzpans kJWVyY"
      >
        <div
          className="sc-fzplWN fCgSYo"
        >
          <div
            className="sc-fznyAO MARrG"
          >
            Mo
          </div>
        </div>
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
      </div>
      <div
        className="sc-fzpans kJWVyY"
      >
        <div
          className="sc-fzplWN fCgSYo"
        >
          <div
            className="sc-fznyAO MARrG"
          >
            Tu
          </div>
        </div>
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
      </div>
      <div
        className="sc-fzpans kJWVyY"
      >
        <div
          className="sc-fzplWN fCgSYo"
        >
          <div
            className="sc-fznyAO MARrG"
          >
            We
          </div>
        </div>
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
      </div>
      <div
        className="sc-fzpans kJWVyY"
      >
        <div
          className="sc-fzplWN fCgSYo"
        >
          <div
            className="sc-fznyAO MARrG"
          >
            Th
          </div>
        </div>
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
      </div>
      <div
        className="sc-fzpans kJWVyY"
      >
        <div
          className="sc-fzplWN fCgSYo"
        >
          <div
            className="sc-fznyAO MARrG"
          >
            Fr
          </div>
        </div>
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
        <div
          className="sc-fzplWN fCgSYo"
        />
      </div>
    </div>
  </div>,
  <div
    className="react-toast-notifications__container css-mrh2d-ToastContainer"
  />,
]
`;
